### ISSUE TITLE
Please add support for retrying

### ISSUE THREAD

**Issue by JasonGross** [JamesIves/github-pages-deploy-action#346]
**Is your feature request related to a problem? Please describe.**

When I use this action on multiple jobs in the same matrix, I don't have a way of ensuring that they don't collide mid-push.

**Describe the solution you'd like**

I'd like to have an option (e.g., `RETRY_COUNT`) that I can set to a number, so that if I set it to the number *N*, then, if the deploy step fails, then the entire deployment is tried again.  (Alternatively, I'd be fine with the action doing a `git pull --rebase` if the `push` fails, and tries this sequence *N* times.)  Then I could set *N* to the number of jobs in the matrix, and be guaranteed that eventually all the pushes would succeed (because on each retry, at least one of the jobs would win the race).

**Comment by JamesIves** [comment]
Could you not just make your jobs dependent on each other like the integration tests work in this repo? Asynchronously deploying through Git is bound to result in failures.

**Comment by JasonGross** [comment]
Not easily.  There is really only one job (the action is [here](https://github.com/coq-community/coq-performance-tests/blob/master/.github/workflows/coq.yml)), which is configured via `strategy: matrix`:
```
    strategy:
      matrix:
        env:
        - { COQ_VERSION: "master", COQ_PACKAGE: "coq        libcoq-ocaml-dev"       , PPA: "ppa:jgross-h/coq-master-daily"             , TIMED: "1", SLEEP_TIME: "1" }
        - { COQ_VERSION: "8.11.2", COQ_PACKAGE: "coq-8.11.2 libcoq-8.11.2-ocaml-dev", PPA: "ppa:jgross-h/many-coq-versions-ocaml-4-05" , TIMED: "1", SLEEP_TIME: "30" }
        - { COQ_VERSION: "8.10.2", COQ_PACKAGE: "coq-8.10.2 libcoq-8.10.2-ocaml-dev", PPA: "ppa:jgross-h/many-coq-versions-ocaml-4-05" , TIMED: "1", SLEEP_TIME: "60" }
        - { COQ_VERSION: "8.9.1" , COQ_PACKAGE: "coq-8.9.1  libcoq-8.9.1-ocaml-dev" , PPA: "ppa:jgross-h/many-coq-versions"            , TIMED: "1", SLEEP_TIME: "90" }
        - { COQ_VERSION: "8.8.2" , COQ_PACKAGE: "coq-8.8.2  libcoq-8.8.2-ocaml-dev" , PPA: "ppa:jgross-h/many-coq-versions"            , TIMED: "1", SLEEP_TIME: "120" }

    env: ${{ matrix.env }}
```
Furthermore, I expect at some point the non-deployment steps will be sufficiently slow that I will want to run them all in parallel, even if the deployment steps need to run in serial.  Is there a way to put the deployment step into a completely separate job?

> Asynchronously deploying through Git is bound to result in failures.

This is true, but I expect (perhaps wrongly?) that in any race to deploy, at least one thread will win.  This is sufficient to upper-bound the number of failures.

**Comment by JamesIves** [comment]
You could still re-organize your build steps, so each one is dependent on each other. I'm not entirely sure what each of those do, but when `build` is complete, then `build2` can run using the second matrix env utilizing the `needs` [syntax](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds). You could also using the deploy status env variable to trigger subsequent runs of the action should it fail also: https://github.com/JamesIves/github-pages-deploy-action#deployment-status

Adding a retry option for an edge case like this is something I'd like to avoid as it adds unnecessary complexity to the package, my goal is to keep it pretty simple. If it fails then it's probably failing for a good reason, and retrying 9/10 times won't work for most users.

**Comment by JasonGross** [comment]
Ah, indeed, I have set up [my workflow](https://github.com/coq-community/coq-performance-tests/blob/51f01995f92682b4230f57e338b260235f4ffa61/.github/workflows/coq.yml#L95-L112) to use a separate job for deployment, which fetches the files to deploy via artifacts.  I ran into another edge-case while doing so, but I'll report a separate issue for that.
